samples/embedded-map/src/components/EmbeddedMap/EmbeddedMap.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useEffect } from "react";
+import {
+    LayoutElement,
+    LayoutElementProperties,
+} from "@vertigis/web/components";
+import EmbeddedMapModel from "./EmbeddedMapModel";
+import { useWatchAndRerender } from "@vertigis/web/ui";
+// Import the necessary CSS for the Mapillary viewer to be styled correctly.
+import "mapillary-js/dist/mapillary.min.css";
+
+export default function EmbeddedMap(
+    props: LayoutElementProperties<EmbeddedMapModel>
+): React.ReactElement {
+    const { model } = props;
+    const { initializeEmbeddedMap, destroyEmbeddedMap, map } = model;
+
+    useWatchAndRerender(model, "map");
+
+    useEffect(() => {
+        if (!map) {
```

Comment:
The framework would never render the React component without a required dependency, so this check isn't really needed.